src/bp/ui-studio/src/web/reducers/flows.ts

```diff
@@ -522,14 +528,59 @@ reducer = reduceReducers(
         flowsByName: doDeleteFlow({ name, flowsByName: state.flowsByName })
       }),
 
+      [requestRefreshCallerFlows]: (state: FlowReducer, { payload }) => {
+        const activeFlow = payload || state.currentFlow
+        const currentFlow = state.flowsByName[activeFlow]
+
+        const outcomeNodes = currentFlow.nodes.filter(x => ['success', 'failure'].includes(x.type))
+        const outcomes = _.orderBy(outcomeNodes, 'type', 'desc').map(x => ({
+          condition: `lastNode=${x.name}`,
+          caption: x.friendlyName || x.name,
+          node: ''
+        }))
+
+        const updatedFlows = {}
+        for (const [name, flow] of Object.entries(state.flowsByName)) {
+          if (!flow.nodes.find(n => n.flow === activeFlow)) {
+            continue
+          }
+
+          const updatedFlow = {
+            ...flow,
+            nodes: flow.nodes.map(node => {
+              if (node.flow !== activeFlow) {
+                return node
+              }
+
+              return {
+                ...node,
+                next: outcomes.map(o => ({ ...o, node: node.next.find(n => n.condition === o.condition)?.node || '' }))
+              }
+            })
+          }
+
+          updatedFlows[name] = updatedFlow
+        }
+
+        return {
+          ...state,
+          flowsByName: {
+            ...state.flowsByName,
+            ...updatedFlows
+          }
+        }
+      },
+
       // Inserting a new skill essentially:
       // 1. creates a new flow
       // 2. creates a new "skill" node
       // 3. puts that new node in the "insert buffer", waiting for user to place it on the canvas
-      [requestInsertNewSkill]: (state, { payload }) => {
+      [requestInsertNewSkill]: (state: FlowReducer, { payload }) => {
         const skillId = payload.skillId
         const flowRandomId = prettyId(6)
-        const flowName = `skills/${skillId}-${flowRandomId}.flow.json`
+
+        const flowPrefix = window.USE_ONEFLOW ? `${parseFlowName(state.currentFlow).workflowPath}/` : ''
+        const flowName = `${flowPrefix}skills-${skillId}-${flowRandomId}.flow.json`
```

Comment:
Not sure if this change is still necessary, it was made so skills were saved under the name of its parent workflow instead of in the skills folder. @slvnperron do we keep skills in the same location as before?

Reply:
I removed this because it's out of scope of this PR